### PR TITLE
Array Inspector utility

### PR DESCRIPTION
**Pull request by DSDubit**

### CODE REVIEW

File: UI/Editor/ArrayInspector.cs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+			{
+				arrayFoldoutBitmask = DrawArray(serializedArray, drawArrayElement, arrayFoldoutBitmask);
+			}
+
+			serializedObject.ApplyModifiedProperties();
+		}
+
+		public static int DrawArray(SerializedProperty arrayProperty, Action<SerializedProperty> drawArrayElement, int selectionBitmask = -1)
+		{
+			if (arrayProperty == null || arrayProperty.arraySize < 0)
```
Comment (HarryHodgson) [inline UI/Editor/ArrayInspector.cs]: Should be either same line or with a block

File: UI/Editor/ArrayInspector.cs [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+					DrawElementContainer(arrayProperty, elementProperty, index, drawArrayElement);
+				}
+			}
+
+			return selectionBitmask;
+		}
+
+		private static void DrawElementContainer(SerializedProperty arrayProperty, SerializedProperty elementProperty, int index, Action<SerializedProperty> drawArrayElement)
+		{
+			var buttonWidth = 60f;
```
Comment (HarryHodgson) [inline UI/Editor/ArrayInspector.cs]: const

File: UI/Editor/ArrayInspector.cs [hunk 3]
```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using UnityEditor;
+using UnityEngine;
+
+namespace Dubit.DUCK.Editor.Inspectors
+{
+	public abstract class ArrayInspector : UnityEditor.Editor
+	{
+		protected abstract string arrayPropertyName
```
Comment (HarryHodgson) [inline UI/Editor/ArrayInspector.cs]: Properties can be one line

File: UI/Editor/ArrayInspector.cs (hunk 1 above)
Comment (DSDubit) [reply UI/Editor/ArrayInspector.cs]: Done

File: UI/Editor/ArrayInspector.cs (hunk 2 above)
Comment (DSDubit) [reply UI/Editor/ArrayInspector.cs]: Done

File: UI/Editor/ArrayInspector.cs (hunk 3 above)
Comment (DSDubit) [reply UI/Editor/ArrayInspector.cs]: Done

**Pull request merged**